src/ts/ContentMetadataBuilder.ts

```diff
@@ -52,47 +35,72 @@ export class ContentMetadataBuilder {
     this.metadataOverrides = { ...this.metadataOverrides, ...newValue };
   }
 
-  getOverrides(): Metadata {
+  getOverrides(): Partial<Metadata> {
     return this.metadataOverrides;
   }
 
   setPlaybackStarted(value: boolean) {
     this.playbackStarted = value;
   }
 
-  build(): Conviva.ConvivaMetadata {
+  private getStaticMetadata() {
+    const metadata: Partial<Metadata> = {};
+
+    // This metadata can only be changed before the playback is started
     if (!this.playbackStarted) {
       // Asset name is only allowed to be set once
-      if (!this.contentMetadata.assetName) {
-        this.contentMetadata.assetName = this.assetName;
-      }
+      metadata.assetName = this.latestBuiltMetadata.assetName || this.assetName;
 
-      this.contentMetadata.viewerId = this.viewerId;
-      this.contentMetadata.streamType = this.metadataOverrides.streamType || this.metadata.streamType;
-      this.contentMetadata.applicationName = this.metadataOverrides.applicationName || this.metadata.applicationName;
-      this.contentMetadata.duration = this.metadataOverrides.duration || this.metadata.duration;
+      metadata.viewerId = this.viewerId;
+      metadata.streamType = this.metadataOverrides.streamType || this.metadata.streamType;
+      metadata.applicationName = this.metadataOverrides.applicationName || this.metadata.applicationName;
+      metadata.duration = this.metadataOverrides.duration || this.metadata.duration;
 
-      this.contentMetadata.custom = this.custom;
+      metadata.custom = this.custom;
     }
+    // If the playback has been started, the values cannot be changed and the latest values before the playback started has to be used
```

Comment:
Comment should be moved into the `else` block

Reply:
Do you mean this way? https://github.com/bitmovin/bitmovin-player-web-analytics-conviva/pull/106/commits/70cad66a428f20a7bfbb2d0616c9387fe878b848

Reply:
Yes, thanks 👍